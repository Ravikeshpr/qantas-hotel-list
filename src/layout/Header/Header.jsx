import React from "react";
import qantasLogo from "../../assets/qantas-logo.png";
import { HeaderContainer, Logo } from "./Header.style";

const Header = () => {
    return (
        <HeaderContainer role="banner">
            <Logo tabIndex={0} src={qantasLogo} alt="Qantas logo" />
        </HeaderContainer>
    );
};

export default Header;
