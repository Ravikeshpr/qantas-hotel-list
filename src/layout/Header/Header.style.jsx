import styled from "styled-components";

export const HeaderContainer = styled.header`
    display: flex;
    flex-wrap: wrap;
    justify-content: start;
    align-items: center;
    padding: 3% 3% 1% 2%;
`;

export const Logo = styled.img`
    width: ${(props) => props.width || "20%"};
    min-width: 100px;
    height: auto;
`;
