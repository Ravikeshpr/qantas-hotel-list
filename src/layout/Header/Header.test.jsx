import React from "react";
import { render, screen } from "@testing-library/react";
import Header from "./Header";
import { describe, it, expect } from "vitest";
import "@testing-library/jest-dom/vitest";

describe("Header", () => {
    it("Header component should have brand logo", () => {
        render(<Header />);
        expect(screen.getByAltText("Qantas logo")).toBeInTheDocument();
    });

    it("Header component should maintain the structure", () => {
        const { container } = render(<Header />);
        expect(container).toMatchSnapshot();
    });
});
