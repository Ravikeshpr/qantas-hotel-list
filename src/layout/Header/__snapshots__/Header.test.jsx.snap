// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > Header component should maintain the structure 1`] = `
<div>
  <header
    class="sc-VHjGu jFIdmD"
    role="banner"
  >
    <img
      alt="Qantas logo"
      class="sc-gnOvAp kcDHvS"
      src="/src/assets/qantas-logo.png"
      tabindex="0"
    />
  </header>
</div>
`;
